test/e2e/install_test.go

```diff
@@ -63,6 +44,8 @@ var _ = Describe("Operator Install", func() {
 		err := c.Create(ctx, operator)
 		Expect(err).ToNot(HaveOccurred())
 
+		// TODO dfranz: This test currently relies on the hard-coded CatalogSources found in bundle_cache.go
```

Comment:
Shouldn't this comment be removed instead of added back since the hard-coded catalog sources are gone?

Reply:
Consolidating comments about e2e here. 

@stevekuznetsov : "Looks like an OLMv0 end-to-end test was removed - is there a new one for this integration?"

I realized after removing the hardcoded entity source that the v0 CatalogSource integration was using the v0 OLM installation to create a CatalogSource with an [index image](https://github.com/operator-framework/operator-controller/blob/4d6757a2b0fe7dda779d03d44f37d20872218711/test/e2e/catalogsource_install_test.go#L78), but then for us to include prod grade e2e tests we probably have to set up the e2e test infrastructure first. That means
* Discuss if we want to install catalogd v0.0.1 as part of the operator-controller installation 
* Set up test infrastructure like a mock image registry, that can be used to build and post an index image into so that the tests can pull from it etc. 

My first instinct is to add the hardcoded entitySource back, so that the deppy tests can go on with the hardcoded entitySource, and we don't increase the scope of this PR with "set up e2e test infrastructure for this project", but then if we feel like we must do it as a part of this PR then I'm happy to do it, with an agreement that it'll increase the scope of this PR. 

cc: @joelanford

Reply:
> Discuss if we want to install catalogd v0.0.1 as part of the operator-controller installation

My guess for this is "yes we want this to be included in this PR, just like we install rukpak", but I'll wait for others to chime in first, to make sure we're comfortable including v0.1.0 release of catalogd as a part of the operator-contorller installation

Reply:
> Discuss if we want to install catalogd v0.0.1 as part of the operator-controller installation

Yes, 100%.

> Set up test infrastructure like a mock image registry, that can be used to build and post an index image into so that the tests can pull from it etc.

Agree with doing this separately. There's a whole bunch wrapped up in this question:
1. Should we do the `catalog.spec.source` union type first so that we don't need all the image infrastructure everywhere we want to test it?
2. What tests should we put in `catalogd` vs `operator-controller`?
3. What's the appropriate layer for the tests? Unit or e2e? Or both?